doctr/travis.py

```diff
@@ -270,3 +274,27 @@ def push_docs():
         run(['git', 'push', '-q', 'doctr_remote', 'gh-pages'])
     else:
         print("The docs have not changed. Not updating")
+
+
+DOCS_REGEX = re.compile('([\w.]*)\/conf\.py')
+
+def find_build_dir():
+    """
+    Locate the local ``Sphinx`` directory and the name of the ``build``
+    directory within it (by default either ``build`` or ``_build``)
+
+    Searches recursively from the root directory for a ``conf.py`` file and
+    then backs out the name of the docs' build folder from there.
+    """
+    for conf_loc in glob.iglob('**/conf.py', recursive=True):
+        doc_loc = re.search(DOCS_REGEX, conf_loc)
+        if doc_loc:
+            doc_loc = doc_loc.group(1)
+            try:
+                build_loc = glob.glob('{}/**build'.format(doc_loc),
+                                      recursive=True)[0]
+            except IndexError:
+                sys.exit('Cannot find build directory.  Is there a ``make html`` in your ``.travis.yml``?')
```

Comment:
I'll change it to a normal exception -- not sure about what to do for an error message.  It fails if there is no `build` directory, which usually means that sphinx hasn't run yet